packages/web-components/fast-foundation/src/data-grid/data-grid.ts

```diff
@@ -601,10 +607,22 @@ export class DataGrid extends FoundationElement {
     };
 
     private updateRowIndexes = (): void => {
-        const newGridTemplateColumns =
-            this.gridTemplateColumns === undefined
-                ? this.generatedGridTemplateColumns
-                : this.gridTemplateColumns;
+        let newGridTemplateColumns = this.gridTemplateColumns;
+
+        if (newGridTemplateColumns === undefined) {
+            // try to generate columns based on manual rows
+            if (this.generatedGridTemplateColumns === "" && this.rowElements.length > 0) {
+                const firstRow = this.rowElements[0] as DataGridRow;
+
+                firstRow.cellElements.forEach(() => {
+                    this.generatedGridTemplateColumns = `${
+                        this.generatedGridTemplateColumns
+                    }${this.generatedGridTemplateColumns === "" ? "" : " "}${"1fr"}`;
+                });
```

Comment:
You could create a new array with "1fr" for each cellElement and create a string using join.
```suggestion
               this.generatedGridTemplateColumns = new Array(this.rowElements[0].length).fill("1fr").join(" ");
```

Reply:
done